### PR TITLE
[@type/node] options.filter of fs.cp() can return a Promise

### PR DESCRIPTION
**Pull request by FuuuOverclocking**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/api/fs.html#fscpsrc-dest-options-callback
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

`options.filter` of `fs.cp(src, dest[, options], callback)` and `fsPromises.cp(src, dest[, options])` can return a Promise that resolves to true or false, according to [documents](https://nodejs.org/api/fs.html#fspromisescpsrc-dest-options).

Also change the arguments' names to match the ones in jsdoc.

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

File: types/node/fs.d.ts [hunk 1]
```diff
@@ -3792,6 +3792,42 @@ declare module 'fs' {
...
+         * be preserved.
+         * @default false
+         */
+        preserveTimestamps?: boolean;
+        /**
+         * Copy directories recursively.
+         * @default false
+         */
+        recursive?: boolean;
+    }
```
Comment (yoursunny) [inline types/node/fs.d.ts:3829]: Missing `verbatimSymlinks`.

File: types/node/fs.d.ts [hunk 2]
```diff
@@ -3792,6 +3792,42 @@ declare module 'fs' {
...
+    export interface CopySyncOptions {
```
Comment (yoursunny) [inline types/node/fs.d.ts:3830]: Maybe extract the common options to `CopyOptionsCommon` to reduce duplication?

**Comment by FuuuOverclocking** [comment]
@yoursunny Thanks for your correction. Now interface `CopyOptionsBase`  includes `verbatimSymlinks`, while `CopyOptions` and `CopySyncOptions` extends it.

**Review by yoursunny — APPROVED** [review]

**Comment by FuuuOverclocking** [comment]
Hello, is anyone here? Please let me know if there is any problem with PR or merge.

**Comment by FuuuOverclocking** [comment]
@peterblazejewicz  

Now the changes have also been applied to v16.

The failure of CI seems to be caused by the new version 4.8 of TypeScript. It has nothing to do with this PR.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by peterblazejewicz — APPROVED** [review]

**Review by mcollina — APPROVED** [review]
lgtm

**Review by gabritto — APPROVED** [review]

**Comment by FuuuOverclocking** [comment]
Ready to merge

**Pull request merged**